sdks/keyvault/src/client.rs

```diff
@@ -127,42 +58,31 @@ impl<'a> KeyVaultClient<'a> {
     ///
     /// ```no_run
     /// use azure_keyvault::KeyVaultClient;
-    /// let client = KeyVaultClient::new(&"c1a6d79b-082b-4798-b362-a77e96de50db", &"SUPER_SECRET_KEY", &"bc598e67-03d8-44d5-aa46-8289b9a39a14", &"test-keyvault");
+    /// use azure_auth_aad::DefaultCredential;
+    /// let creds = DefaultCredential::default();
+    /// let client = KeyVaultClient::new(&creds, &"test-keyvault");
     /// ```
-    pub fn new(
-        aad_client_id: &'a str,
-        aad_client_secret: &'a str,
-        aad_tenant_id: &'a str,
-        keyvault_name: &'a str,
-    ) -> Self {
+    pub fn new(token_credential: &'a T, keyvault_name: &'a str) -> Self {
         KeyVaultClient::with_endpoint_suffix(
-            aad_client_id,
-            aad_client_secret,
-            aad_tenant_id,
+            token_credential,
             keyvault_name,
             PUBLIC_ENDPOINT_SUFFIX.to_owned(),
         )
     }
 
     pub(crate) async fn refresh_token(&mut self) -> Result<(), KeyVaultError> {
-        if matches!(self.token_expiration, Some(exp) if exp > chrono::Utc::now()) {
+        if matches!(&self.token, Some(token) if token.expires_on > chrono::Utc::now()) {
```

Comment:
Just curious, should we provide some sort of buffer where we refresh the token even if its technically still valid? If the token is valid for 1ns more we won't refresh even though we arguably should.

Reply:
I made no logic change here either. It was the original implementation.